src/main/java/org/grouplens/grapht/reflect/internal/ClassInstantiator.java

```diff
@@ -98,38 +107,59 @@ public Object instantiate() throws ConstructionException {
         } catch (IllegalAccessException e) {
             throw new ConstructionException(ctor, "Access violation on " + ctor, e);
         }
+        finally {
+            mdcContext.finish();
+        }
 
         // satisfy dependencies in the order of the list, which was
         // prepared to comply with JSR 330
         Map<Method, InjectionArgs> settersAndArguments = new HashMap<Method, InjectionArgs>();
         for (Desire d: desires) {
+            Object value;
+            FieldInjectionPoint fd = null;
+            Field field;
             if (d.getInjectionPoint() instanceof FieldInjectionPoint) {
-                FieldInjectionPoint fd = (FieldInjectionPoint) d.getInjectionPoint();
-                Object value = checkNull(fd, providers.get(d).instantiate());
-                Field field = fd.getMember();
-
+                LogContext mdcContextInjectionPtField =  LogContext.create();
                 try {
+                    fd = (FieldInjectionPoint) d.getInjectionPoint();
+                    value = checkNull(fd, providers.get(d).instantiate());
+                    field = fd.getMember();
                     logger.trace("Setting field {} with arguments {}", field, value);
                     field.setAccessible(true);
                     field.set(instance, value);
+                    mdcContextInjectionPtField.put("org.grouplens.grapht.injectionPoint", fd.toString());
```

Comment:
We only need this one, not the two following ones (`fd.toString()` includes the relevant values).

Also, it needs to be at the beginning of the `try`.